test/framework/machinepool_helpers.go

```diff
@@ -232,39 +240,28 @@ func WaitForMachinePoolInstancesToBeUpgraded(ctx context.Context, input WaitForM
 type GetMachinesPoolInstancesInput struct {
 	Getter      Getter
 	Namespace   string
-	MachinePool clusterv1exp.MachinePool
+	MachinePool *clusterv1exp.MachinePool
 }
 
-// GetMachinePoolInstanceVersions returns the.
-func GetMachinePoolInstanceVersions(ctx context.Context, input GetMachinesPoolInstancesInput) []string {
-	Expect(ctx).NotTo(BeNil(), "ctx is required for GetMachinePoolInstanceVersions")
-	Expect(input.Namespace).ToNot(BeEmpty(), "Invalid argument. input.Namespace can't be empty when calling GetMachinePoolInstanceVersions")
-	Expect(input.MachinePool).ToNot(BeNil(), "Invalid argument. input.MachineDeployment can't be nil when calling GetMachinePoolInstanceVersions")
-
-	obj := getUnstructuredRef(ctx, input.Getter, &input.MachinePool.Spec.Template.Spec.InfrastructureRef, input.Namespace)
-	instances, found, err := unstructured.NestedSlice(obj.Object, "status", "instances")
-	Expect(err).ToNot(HaveOccurred(), "failed to extract machines from unstructured")
-	if !found {
-		return nil
-	}
+// getMachinePoolInstanceVersions returns the Kubernetes versions of the machine pool instances.
+func getMachinePoolInstanceVersions(ctx context.Context, input GetMachinesPoolInstancesInput) []string {
+	Expect(ctx).NotTo(BeNil(), "ctx is required for getMachinePoolInstanceVersions")
+	Expect(input.Getter).ToNot(BeNil(), "Invalid argument. input.Getter can't be nil when calling getMachinePoolInstanceVersions")
+	Expect(input.Namespace).ToNot(BeEmpty(), "Invalid argument. input.Namespace can't be empty when calling getMachinePoolInstanceVersions")
+	Expect(input.MachinePool).ToNot(BeNil(), "Invalid argument. input.MachineDeployment can't be nil when calling getMachinePoolInstanceVersions")
 
+	instances := input.MachinePool.Status.NodeRefs
 	versions := make([]string, len(instances))
 	for i, instance := range instances {
-		version, found, err := unstructured.NestedString(instance.(map[string]interface{}), "version")
-		Expect(err).ToNot(HaveOccurred(), "failed to extract versions from unstructured instance")
-		Expect(found).To(BeTrue(), "unable to find nested version string in unstructured instance")
-		versions[i] = version
+		node := &corev1.Node{}
+		err := input.Getter.Get(ctx, client.ObjectKey{Name: instance.Name}, node)
```

Comment:
I need to double check this in the logs, but I'm not sure this could work in the current form because the Getter is reading from the management cluster, while those nodes exists in the workload cluster.

Reply:
I'm passing in a workload cluster Getter now.